### PR TITLE
Various test fixes, enable linting in CI, add Python 3.7 compatibility

### LINKED ISSUE #77: Intermittent test failures make development difficult
Adding features to `gitlint` (or improving existing ones) is unfortunately pretty difficult due to intermittently failing integration tests. This can also be seen in simple pull requests such as #73.

I tried a few things in #76, but haven't been able to fix it so far.

My feeling is that it is related to unicode string handling, possibly also in connection with an outdated version of `sh` that returns the strings that are compared to the reference result. The [sh changelog](https://github.com/amoffat/sh/blob/master/CHANGELOG.md) shows quite a few unicode-related bugs being fixed in versions newer than the one currently used by `gitlint`. Those newer versions appear to be incompatible with `gitlint` today, so simply updating the dependency is not enough.

### PR DESCRIPTION
**Pull request by pbregener**
This changes various dependencies and test-related files.
* Updates the `pylint` dependency which allows us to run the linting tests with Python 3.6
* Fixes a few warnings emitted after updating `pylint`
* Updates other dependencies such as `sh` and `pytest` (needed for Python 3.7 compatibility)
* Adds Travis CI tests using Python 3.7
* Fixes intermittent test failures in Travis CI (fixes #77)
* Enables all tests for all Python versions in Travis CI
* The failing test for Python 3.7 (using an Ubuntu Xenial image) should be fixed by #79

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,45 +9,26 @@ matrix:
       os: linux
     - python: "3.3"
       os: linux
-      script:
```
[the remark this answers is not in the document]
Comment (pbregener) [reply .travis.yml]: I disabled them for Python 3.3-3.7 and pypy now. (So they are only running for Python 2.6-2.7.)

**Comment by pbregener** [comment]
Now one integration test even failed for Python 2.6. Should we just globally disable the integration tests in Travis for now?

**Comment by pbregener** [comment]
This still needs a proper cleanup, but I think I fixed the main integration test issues @jorisroovers! Instead of using `assertEqual` on the returned object from the `sh` call (which is not a string), a new function `assertEqualStdout` takes care of some unicode and linebreak issues.

The one Python 3.7 unit test failure on Travis seems to be reproducible (on Travis), but I cannot reproduce it locally using Python 3.7.0.

I will not have time to work on this the next days or the next week, but feel free to cherry-pick from this, or finish this PR yourself :)

**Comment by pbregener** [comment]
Could you review this again @jorisroovers? I think this finally fixes the intermittent failures for all Python versions! Tests for Python 3.7 fail due to a newer git version included in the image used by Travis (but PR #79 fixes that).

**Comment by pbregener** [comment]
ping @jorisroovers

**Comment by jorisroovers** [comment]
@pbregener sorry, it's really hard for me to make time to work on this these days. Happy to take your word for it and get this merged for now. 

I don't anticipate having proper time to review this, do some of the other work and then push a new release until late December when I will take a holiday though.

If you're interested in taking a more active/overall role on the project (and pinky swear to uphold proper standards ;-) ), I can give you full commit access. Let me know!

**Comment by pbregener** [comment]
Appreciate it @jorisroovers, but I am about to start a new job soon and am therefore pretty busy, too, these days. I have just found gitlint quite useful and created this PR with the hope to help make it more future proof :)

**Pull request merged**